File: budget/src/main/java/com/taubel/budget/entities/Category.java
package com.taubel.budget.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Categories")
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "category_id")
    private Long id;

    private String name;

    @ManyToOne()
    @JoinColumn(name = "budget_id")
    private Budget budget;

    @ManyToOne()
    @JoinColumn(name = "user_id")
    private User user;  
    
}

File: FerengiFin/src/app/shared/models/category.ts
export class Category {
    id: number
    name: string
    userId: number
    budgetId: number

  constructor(
        id: number, 
        name: string, 
        userId: number, 
        budgetId: number
    ) {
        this.id = id
        this.name = name
        this.userId = userId
        this.budgetId = budgetId
    }
}